From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Patrick McCarty <patrick.mccarty@intel.com>
Date: Sun, 1 Nov 2020 17:31:04 +0000
Subject: [PATCH] Install and process auxiliary libtool m4 files

The libtool.m4 distributed in the php tree appears to be custom, not
requiring the other, auxiliary m4 files needed by the upstream
libtool.m4.

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 scripts/Makefile.frag | 4 ++++
 scripts/phpize.in     | 2 +-
 scripts/phpize.m4     | 4 ++++
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/scripts/Makefile.frag b/scripts/Makefile.frag
index bfbac62a..8bbc824e 100644
--- a/scripts/Makefile.frag
+++ b/scripts/Makefile.frag
@@ -8,6 +8,10 @@ phpbuilddir = $(libdir)/build
 BUILD_FILES = \
 	scripts/phpize.m4 \
 	build/libtool.m4 \
+	build/ltoptions.m4 \
+	build/ltsugar.m4 \
+	build/ltversion.m4 \
+	build/lt~obsolete.m4 \
 	build/ltmain.sh \
 	build/ax_check_compile_flag.m4 \
 	build/ax_gcc_func_attribute.m4 \
diff --git a/scripts/phpize.in b/scripts/phpize.in
index 04864733..42a070a8 100644
--- a/scripts/phpize.in
+++ b/scripts/phpize.in
@@ -9,8 +9,8 @@ includedir="`eval echo @includedir@`/php"
 builddir="`pwd`"
 SED="@SED@"
 
-FILES_BUILD="php.m4 shtool libtool.m4 ax_check_compile_flag.m4 ax_gcc_func_attribute.m4 php_cxx_compile_stdcxx.m4 pkg.m4 \
-    config.guess config.sub ltmain.sh Makefile.global gen_stub.php"
+FILES_BUILD="php.m4 shtool libtool.m4 ax_check_compile_flag.m4 ax_gcc_func_attribute.m4 php_cxx_compile_stdcxx.m4 pkg.m4 \
+    config.guess config.sub ltmain.sh Makefile.global gen_stub.php ltoptions.m4 ltsugar.m4 ltversion.m4 lt~obsolete.m4"
 FILES="run-tests*.php"
 CLEAN_FILES="$FILES *.o *.lo *.la .libs/ build/ modules/ \
 	config.nice configure configure.ac \
diff --git a/scripts/phpize.m4 b/scripts/phpize.m4
index cdd0e129..09fa7f95 100644
--- a/scripts/phpize.m4
+++ b/scripts/phpize.m4
@@ -5,6 +5,10 @@ dnl comments starting with # and empty newlines from the included files.
 m4_include([build/ax_check_compile_flag.m4])
 m4_include([build/ax_gcc_func_attribute.m4])
 m4_include([build/libtool.m4])
+m4_include([build/ltoptions.m4])
+m4_include([build/ltsugar.m4])
+m4_include([build/ltversion.m4])
+m4_include([build/lt~obsolete.m4])
 m4_include([build/php_cxx_compile_stdcxx.m4])
 m4_include([build/php.m4])
 m4_include([build/pkg.m4])
